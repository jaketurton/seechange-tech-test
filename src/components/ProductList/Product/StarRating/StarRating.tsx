interface StarRatingProps {
  rating: number;
}

export default function StarRating({ rating }: StarRatingProps) {
  const normalizedRating = Math.max(0, Math.min(5, rating));

  const stars = [];
  for (let i = 1; i <= 5; i++) {
    stars.push(i <= normalizedRating ? "â˜…" : "");
  }

  return (
    <div className="mb-2">
      {stars.map((star, index) => (
        <span key={index} className="text-yellow-400 font-black text-base mr-1">
          {star}
        </span>
      ))}
    </div>
  );
}
