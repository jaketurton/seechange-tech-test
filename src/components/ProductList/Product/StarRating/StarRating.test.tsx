import { render } from "@testing-library/react";
import StarRating from "./StarRating";

describe("StarRating Component", () => {
  test("matches the snapshot", () => {
    const { asFragment } = render(<StarRating rating={3} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("displays correct number of filled stars for a rating", () => {
    const { container } = render(<StarRating rating={3} />);
    expect(container.textContent).toMatch("★★★");
  });

  test("handles ratings below 0 and above 5 correctly", () => {
    const { rerender, container } = render(<StarRating rating={-1} />);
    expect(container.textContent).toMatch("");

    rerender(<StarRating rating={6} />);
    expect(container.textContent).toMatch("★★★★★");
  });
});
