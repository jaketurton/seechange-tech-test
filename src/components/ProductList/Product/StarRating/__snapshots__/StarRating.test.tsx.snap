// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarRating Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mb-2"
  >
    <span
      class="text-yellow-400 font-black text-base mr-1"
    >
      ★
    </span>
    <span
      class="text-yellow-400 font-black text-base mr-1"
    >
      ★
    </span>
    <span
      class="text-yellow-400 font-black text-base mr-1"
    >
      ★
    </span>
    <span
      class="text-yellow-400 font-black text-base mr-1"
    />
    <span
      class="text-yellow-400 font-black text-base mr-1"
    />
  </div>
</DocumentFragment>
`;
